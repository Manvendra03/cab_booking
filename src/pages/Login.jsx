/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import { Circles } from 'react-loader-spinner';
import { loginUser } from '../redux/slices/userSlice';

const Login = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();
  const [loading, setLoading] = useState(false);

  const handleLoginSubmit = async ({ email, password }) => {
    setLoading(true);
    const user = {
      email,
      password,
    };
    const result = await dispatch(loginUser(user));
    if (result?.payload?.status === 200) {
      toast.success('Login successful');
      navigate('/cabs');
      setLoading(false);
    } else {
      navigate('/cabs');
      // toast.error('Login failed, Invalid credentials');
      setLoading(false);
    }
  };

  return (
    <div className="pt-40 space-y-8 bg-lime-50  flex flex-col items-center h-screen overflow-auto">
      <h2 className="flex justify-center text-3xl md:text-4xl tracking-widest text-lime-800 font-bold">
        Login
      </h2>
      <form
        onSubmit={handleSubmit(handleLoginSubmit)}
        className="w-full  min-w-full  md:min-w-[600px] max-w-[800px] grid grid-cols-1 gap-4 mb-20 px-4 md:px-10 text-lime-800 lg:px-20 xl:px-40 md:gap-x-8"
      >
        <div>
          <input
            type="text"
            id="email"
            name="email"
            placeholder="Email"
            {...register('email', {
              required: 'Please enter your email',
              pattern: {
                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                message: 'Please enter a valid email address',
              },
            })}
          />
          {errors.email && (
            <span className="text-red-500">{errors.email.message}</span>
          )}
        </div>
        <div>
          <input
            type="password"
            id="password"
            name="password"
            placeholder="Password"
            {...register('password', {
              required: 'Please enter the Password',
            })}
          />
          {errors.password && (
            <span className="text-red-500 text-base">
              {errors.password.message}
            </span>
          )}
        </div>
        {loading ? (
          <div className="flex justify-center w-full">
            <Circles
              height="50"
              width="50"
              color="rgba(101, 163, 13, 1)"
              ariaLabel="circles-loading"
              wrapperStyle={{}}
              wrapperClass=""
              visible={loading}
            />
          </div>
        ) : (
          <button
            type="submit"
            className="bg-lime-700
            w-full flex gap-2 items-center transition-all duration-300 text-white rounded-md justify-center text-2xl px-8 py-2"
          >
            Login
          </button>
        )}
      </form>
      <p className="flex justify-center">
        Don&apos;t have an account?&nbsp;
        <Link to="/signup" className="text-blue-800 font-italic">
          Sign Up
        </Link>
      </p>
    </div>
  );
};

export default Login;
