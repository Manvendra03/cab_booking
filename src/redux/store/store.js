import { configureStore } from '@reduxjs/toolkit';
import userReducer from '../slices/userSlice';
import fetchCabReducer from '../slices/cabSlice';
import reservationReducer from '../slices/reservationSlice';
import menuReducer from '../slices/menuSlice';

const store = configureStore({
  reducer: {
    fetchCab: fetchCabReducer,
    user: userReducer,
    reservation: reservationReducer,
    menu: menuReducer,
  },
});

export default store;
