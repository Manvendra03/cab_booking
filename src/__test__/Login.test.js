import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store/store';
import Login from '../pages/Login';

describe('Delete Cab', () => {
  it('renders sign up component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Sign up form', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );

    const signup = screen.getByText(/Sign Up/i);
    expect(signup).toBeInTheDocument();
  });
  it('renders sign up component and whows the already have an account test', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );

    const subHead = screen.getByText(/Don't have an account?/i);
    expect(subHead).toBeInTheDocument();
  });
});
