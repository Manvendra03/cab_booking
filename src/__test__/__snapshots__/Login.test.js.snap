// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Cab renders sign up component 1`] = `
<div
  className="pt-40 space-y-8 bg-lime-50 h-screen overflow-auto"
>
  <h2
    className="flex justify-center text-3xl md:text-4xl tracking-widest text-lime-800 font-bold"
  >
    Login
  </h2>
  <form
    className="w-full grid grid-cols-1 gap-4 mb-20 px-4 md:px-10 text-lime-800 lg:px-20 xl:px-40 md:gap-x-8"
    onSubmit={[Function]}
  >
    <div>
      <input
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        type="text"
      />
    </div>
    <div>
      <input
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="bg-lime-700 w-full flex gap-2 items-center transition-all duration-300 text-white rounded-md justify-center text-2xl px-8 py-2"
      type="submit"
    >
      Login
    </button>
  </form>
  <p
    className="flex justify-center"
  >
    Don't have an account?Â 
    <a
      className="text-blue-800 font-italic"
      href="/signup"
      onClick={[Function]}
    >
      Sign Up
    </a>
  </p>
</div>
`;
